src/Umbraco.Web.BackOffice/Security/PasswordChanger.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System;
+using System.ComponentModel.DataAnnotations;
+using System.Threading.Tasks;
+using Umbraco.Core;
+using Umbraco.Core.BackOffice;
+using Umbraco.Core.Logging;
+using Umbraco.Core.Models;
+using Umbraco.Extensions;
+using Umbraco.Web.Models;
+using IUser = Umbraco.Core.Models.Membership.IUser;
+
+namespace Umbraco.Web.BackOffice.Security
+{
+    internal class PasswordChanger
+    {
+        private readonly ILogger _logger;
+
+        public PasswordChanger(ILogger logger)
+        {
+            _logger = logger;
+        }
+
+        /// <summary>
+        /// Changes the password for a user based on the many different rules and config options
+        /// </summary>
+        /// <param name="currentUser">The user performing the password save action</param>
+        /// <param name="savingUser">The user who's password is being changed</param>
+        /// <param name="passwordModel"></param>
+        /// <param name="userMgr"></param>
+        /// <returns></returns>
+        public async Task<Attempt<PasswordChangedModel>> ChangePasswordWithIdentityAsync(
+            IUser currentUser,
+            IUser savingUser,
+            ChangingPasswordModel passwordModel,
+            BackOfficeUserManager userMgr)
+        {
+            if (passwordModel == null) throw new ArgumentNullException(nameof(passwordModel));
+            if (userMgr == null) throw new ArgumentNullException(nameof(userMgr));
+
+            if (passwordModel.NewPassword.IsNullOrWhiteSpace())
+            {
+                return Attempt.Fail(new PasswordChangedModel { ChangeError = new ValidationResult("Cannot set an empty password", new[] { "value" }) });
```

Comment:
Can these strings be localised?

Reply:
I think yes, but not really related to this PR. This is just copied over. We could make an up-for-grabs task about localizing them